cl/citations/filter_parentheticals.py

```diff
@@ -0,0 +1,76 @@
+
+_SURROUNDING_CHARS = r'[.!;,"“” ]'
+
+PARENTHETICAL_REGEX_BLACKLIST_RULES = [
+    r".n banc",  # en banc or in banc
+    f"{_JUDGE_NAME}(?: {_FULL_OPINION_DESCRIPTOR})?(?:[ ,]+(?:and )?{_FULL_OPINION_DESCRIPTOR})*",  # Scalia, J., dissenting; Roberts, C.J., concurring in the judgment, concurring in part, and dissenting in part
+    f"(?:{_DOCUMENT_TYPES} )?{_FULL_OPINION_DESCRIPTOR}",  # concurring in result
+    f"{_DOCUMENT_TYPES} of {_JUDGE_NAME}",  # opinion of Breyer, J.; opinion of Scalia and Alito, J.J.
+    f"{_OPINION_TYPES}(?: {_DOCUMENT_TYPES})?(?: {_OPINION_TYPE_MODIFICATION})?",  # plurality opinion, supplemental order
+    r"dictum|dicta",
+    r"on rehearing|denying cert(iorari)?",
+    r"simplified|cleaned up|as amended",
+    r"same|similar|contra",
+    r"(?:and )?cases cited therein",
+    r"No. \d+.?\d+",  # No. 12-345
+    r"n. \d+",  # n. 6
+    f"{_DOCUMENT_TYPES} below",  # case below
+    f"{_AGGREGATOR_TYPES} {_DOCUMENT_TYPES}s?",  # collecting cases, reviewing cases
+    f"(?:{_OPINION_TYPES} )?table(?: {_DOCUMENT_TYPES})?",
+    f"{_FULL_MODIFIABLE} {_QUOTE_MODIFICATION}",  # internal citations omitted
+    f"{_FULL_MODIFIABLE} and {_FULL_MODIFIABLE} {_QUOTE_MODIFICATION}",
+    f"{_FULL_MODIFIABLE} {_QUOTE_MODIFICATION}[;,] ?{_FULL_MODIFIABLE} {_QUOTE_MODIFICATION}",
+    f"(?:{_MODIFABLE_TYPE} )?{_MODIFIABLE}, {_MODIFIABLE}, and {_MODIFIABLE} {_QUOTE_MODIFICATION}",
+    f"{_REFERENTIAL} .*",  # citing Gonzales v. Raich, 123 U.S. 456 (2019). A tad over-inclusive but very helpful
+    r".{1,20} \d{4}",  # 2nd Cir. 2019, Third Circuit 1993
+    r".{1,25} I+",  # Gonzales II
+    r".{1,10} (?:Circuit|Cir.)",  # Tenth Circuit, 5th Cir.
+    r"here(?:in)?after .+",  # hereinafter, "Griffin II"
+    r"\S*",  # Single-word parentheticals, e.g., 'TILA'
+]
+
+_PREFIX = rf"^{_SURROUNDING_CHARS}*(?:"  # Begin string, optional whitespace/puncutation, begin capture group
+_SUFFIX = rf"){_SURROUNDING_CHARS}*$"  # Close capture group, optional whitespace/punctuation, end string
+
+PARENTHETICAL_BLACKLIST_REGEX = re.compile(
+    _PREFIX
+    + r"|".join(  # Wrap each rule in its own non-capturing group
+        map(lambda reg: f"(?:{reg})", PARENTHETICAL_REGEX_BLACKLIST_RULES)
+    )
+    + _SUFFIX,
+    re.IGNORECASE,
+)
+
+
+def is_parenthetical_descriptive(text: str) -> bool:
+    text = clean_parenthetical_text(text)
+    return not re.match(PARENTHETICAL_BLACKLIST_REGEX, text)
+
+
+def clean_parenthetical_text(text: str) -> str:
+    text = re.sub(
+        r"\* ?\d+", " ", text
+    )  # Remove star page numbers (e.g. *389)
+    text = re.sub(
+        r" [_-]{4,} ", " ", text
+    )  # Remove weird "word ----- word" stuff (but don't screw up something like "123 U.S. ---")
+    text = re.sub(
+        r"\s+", " ", text
+    )  # Remove excess whitespace that may have appeared as a result of sequential dash sequences
+    return text
```

Comment:
Just a style thing, and I know you've satisfied `black`, but I think I'd rather see these comments on their own lines above what they do, rather than trailing after. Don't feel obligated to fix this, but this looks weird to me.